src/relay/op/nn/nn.cc

```diff
@@ -146,6 +146,23 @@ RELAY_REGISTER_UNARY_OP("relay.op.nn._make.", "relu")
 // Dropout
 TVM_REGISTER_NODE_TYPE(DropoutAttrs);
 
+bool DropoutRel(const Array<Type>& types,
+                int num_inputs,
+                const Attrs& attrs,
+                const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 2);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) return false;
+
+  // dropout returns the original tensor with dropout applied
+  // and a mask tensor (1.0 where element not dropped, 0.0 where dropped)
+  std::vector<Type> fields;
+  fields.push_back(TensorTypeNode::make(data->shape, data->dtype));
+  fields.push_back(TensorTypeNode::make(data->shape, data->dtype));
+  reporter->Assign(types[1], TupleTypeNode::make(Array<Type>(fields)));
```

Comment:
You can try using TupleTypeNode::make(Array<Type>({ty, ty}));

Reply:
Duly noted. I've had some type errors with similar constructions, so I'll give it another shot.